### PR TITLE
Fix bullet bug

### PR DESCRIPTION
**Pull request by JiuqingSong**
1. Toggle bullet/numbering for empty line may put cursor in incorrect position in Edge, fix it by adding a space before toggle, and remove it after toggle.

2. Add a function in Editor class to allow suspending undo snapshot.

3. Add git and wiki link to sample site.

### CODE REVIEW

File: packages/roosterjs-editor-api/lib/format/toggleBullet.ts [hunk 1]
```diff
@@ -1,9 +1,36 @@
...
+        }
+    }
+
+    return false;
+}
+
+export function removeWorkaroundForEdge(editor: Editor) {
+    let node = getNodeAtCursor(editor);
+    if (node.nodeType == NodeType.Element && (<Element>node).textContent == ' ') {
+        (<Element>node).textContent = '';
```
Comment (MSJiaGuo) [inline packages/roosterjs-editor-api/lib/format/toggleBullet.ts]: It seems that you did nothing for the node.

Comment (JiuqingSong) [reply packages/roosterjs-editor-api/lib/format/toggleBullet.ts]: I changed its content from ' ' to ''.

File: packages/roosterjs-editor-api/lib/format/toggleBullet.ts [hunk 2]
```diff
@@ -1,9 +1,36 @@
...
+    }
+}
 
 export default function toggleBullet(editor: Editor): void {
     editor.focus();
     execFormatWithUndo(editor, () => {
+        let hasWoraround = workaroundForEdge(editor);
```
Comment (redmondbowang) [inline packages/roosterjs-editor-api/lib/format/toggleBullet.ts]: hasWorkaround

File: packages/roosterjs-editor-core/lib/editor/Editor.ts [hunk 3]
```diff
@@ -57,6 +57,7 @@ export default class Editor {
     private isBeforeDeactivateEventSupported: boolean;
     private undoService: UndoService;
     private isInIMESequence: boolean;
+    private suspendAddingUndoSnapshot: boolean;
```
Comment (redmondbowang) [inline packages/roosterjs-editor-core/lib/editor/Editor.ts]: Is this default to 'false'?

File: packages/roosterjs-editor-api/lib/format/toggleBullet.ts [hunk 4]
```diff
@@ -1,9 +1,36 @@
...
+import getNodeAtCursor from '../cursor/getNodeAtCursor';
+import { Editor, browserData } from 'roosterjs-editor-core';
+import { NodeType } from 'roosterjs-editor-types';
+
+// Edge may incorrectly put cursor after toggle bullet, workaround it by adding a space.
+// The space will be removed by Edge after toggle bullet
+export function workaroundForEdge(editor: Editor): boolean {
+    if (browserData.isEdge) {
+        let node = getNodeAtCursor(editor);
+        if (node.nodeType == NodeType.Element && (<Element>node).textContent == '') {
```
Comment (redmondbowang) [inline packages/roosterjs-editor-api/lib/format/toggleBullet.ts]: If it's an image element, would textContent also be ''? seems to be okay though, just curious.

Comment (JiuqingSong) [reply packages/roosterjs-editor-api/lib/format/toggleBullet.ts]: Good cache. I have made a fix for this scenario.

File: packages/roosterjs-editor-api/lib/format/toggleBullet.ts (hunk 2 above)
Comment (JiuqingSong) [reply packages/roosterjs-editor-api/lib/format/toggleBullet.ts]: Fixed

File: packages/roosterjs-editor-core/lib/editor/Editor.ts (hunk 3 above)
Comment (JiuqingSong) [reply packages/roosterjs-editor-core/lib/editor/Editor.ts]: Yes.

**Pull request merged**